### ISSUE TITLE
Pandora on Raspbmc keeps crashing

### ISSUE THREAD

**Issue by ratherDashing** [rivy/xbmc-script.audio.pandora#6]
I've tried pretty much every Pandora plug-in and they all crash.  Here is a xbmc log on debug with the application crashing.  Not much in terms of information though.  If you want me to tweak some settings let me know.  It crashes within a few songs.

http://pastebin.com/nPJwHB23

**Comment by rivy** [comment]
It's difficult to tease out what's happening out of the debug log.

My initial thought is that it might be the RaspBMC player is faulting on the MP4 audio stream. The current base player in XBMC has difficulty playing an MP4 audio file unless I specify that it should use the music player (via the `xbmc.PLAYER_CORE_MPLAYER` initialization setting; see bug report @ http://trac.xbmc.org/ticket/14854 ).

You might try using the high quality setting (it might require a Pandora ONE account) which will pull down MP3 audio. That may solve the problem, and would point to a player issue.

I'll add some more log instrumentation to the script as well which may help narrow down the problem with another debug run. I'll let you know when I get the updated logging pushed into a release.

**Comment by rivy** [comment]
@ratherDashing,

I refactored the logging code and added several extra logging points. If you'll use the most recent blead branch commit ( c7aafd8d9ed2850 ), we might be able to narrow the problem down a bit more.

Did you get a chance to test the code using MP3 streams (High quality)?

**Comment by ratherDashing** [comment]
@rivy ,

Thanks for adding that additional logging.  My wife gave birth yesterday so we are both still at the hospital.  Within the next few days I'll be able to test.

I won't be able to to test the high quality stream as I do not have a Pandora One account.

**Comment by robrocke** [comment]
I've been having the same problem for some time. I've tried both Raspbmc and OpenELEC and on both the Pandora add-on seems unstable. I am able to start the Pandora add-on and log in with my stored Pandora account credentials successfully. Sometimes it then immediately crashes when trying to play the first song, sometimes it will play a couple of songs, but always it eventually crashes -- normally sooner rather than later. This behavior is the same even if the only thing I try and do after a fresh reboot is to start the Pandora add-on. I have the add-on set to "High" audio quality. Anything I can do to help troubleshoot?

**Comment by rivy** [comment]
@ratherDashing, Congratulations! Enjoy your new family!

@robrocke, if you could run https://github.com/rivy/xbmc-script.audio.pandora/commit/c7aafd8d9ed2850 in debug mode and link a pastebin of it, that might help narrow down the issue. Right now, I just don't have enough information to be able to say whether this addon is breaking or the XBMC player is choking on the stream.

**Comment by ratherDashing** [comment]
http://pastebin.com/kqMNxSVy

hope I updated the plug-in correctly!

**Comment by rivy** [comment]
@ratherDashing ,

Yes, it looks like you did install it correctly. And from the log you posted, it looks like the player read four songs into the playlist, played three, started the fourth which triggers a request for more songs, and then the script ends without posting any closing notation. So, it might have crashed at  that point.

Does that sound like what happened during the session?

If so, it may be an error in the playlist logic for adding songs.

I expected to see some sort of exception in the log, but I didn't. So, I'll add some try/except blocks in the code to capture and report any currently unseen exceptions. I'll create another commit with further debug logic for you and post the commit ID in this thread.

As I don't have a RaspBMC player, I can't really track this down without help.

So, thanks for helping out.

@robrocke , did you have any luck installing the commit and executing it? If so, does the description above match your experience with it?

**Comment by rivy** [comment]
I've wrapped the addon in a try/catch block, attempting to ferret out any silent exceptions. I would have expected XBMC to log them, but I don't see it. The script is exiting without logging any cleanup activity which appears to me to be an unhandled exception despite the lack of a log entry to that effect.

I've also increased the logging instrumentation in the getMoreSongs() function, trying to narrow down focus on which code is being troublesome.

The commit containing all of this is c8ea09be14c20ffc894c0b8e2ff67adc9ca19d03; the zip is @ https://github.com/rivy/xbmc-script.audio.pandora/archive/c8ea09be14c20ffc894c0b8e2ff67adc9ca19d03.zip . If you can install it, run in debug, and give me some feedback (including the debug log and what happened during the run [such as played x songs, then crashed, etc.]), I'd appreciate it.

**Comment by ratherDashing** [comment]
@rivy 

Here you go! http://paste2.org/IIawA4zj
